### PR TITLE
Update BeatmapOverlay header in line with the web design

### PR DESCRIPTION
**Pull request by EVAST9919**
Screenshot_1

I tried to make changes as little as possible.

### CODE REVIEW

File: osu.Game/Overlays/OverlayRulesetTabItem.cs [hunk 1]
```diff
@@ -64,29 +64,33 @@ public OverlayRulesetTabItem(RulesetInfo value)
...
-        protected override void OnDeactivated() => updateState();
+        protected override void OnDeactivated() => UpdateState();
 
-        private void updateState()
+        protected void UpdateState()
         {
             text.Font = text.Font.With(weight: Active.Value ? FontWeight.Bold : FontWeight.Medium);
+
+            if (!Enabled.Value)
+                return;
```
Comment (bdach) [inline osu.Game/Overlays/OverlayRulesetTabItem.cs]: I don't understand what's going on with `UpdateState()` - why is it exposed as protected and why does it have this guard in the middle?

File: osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs [hunk 2]
```diff
@@ -3,143 +3,87 @@
...
+                    UpdateState();
+                    return;
+                }
 
-        protected override bool OnHover(HoverEvent e)
-        {
-            updateState();
-            return false;
+                AccentColour = colourProvider.Foreground1;
+            }, true);
```
Comment (bdach) [inline osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs]: I'm also really confused by this part too. I can't tell what this is trying to achieve or why `AccentColour` is set here, and I don't think I should be digging in this code to find out.

File: osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs [hunk 3]
```diff
@@ -3,143 +3,87 @@
...
+        {
+            base.LoadComplete();
 
             BeatmapSet.BindValueChanged(setInfo =>
             {
                 var beatmapsCount = setInfo.NewValue?.Beatmaps.Count(b => b.Ruleset.Equals(Value)) ?? 0;
 
                 count.Text = beatmapsCount.ToString();
-                count.Alpha = beatmapsCount > 0 ? 1f : 0f;
+                countContainer.FadeTo(beatmapsCount > 0 ? 1 : 0);
```
Comment (bdach) [inline osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs]: Kind of a strange change, as there doesn't seem to be any functional difference between `FadeTo` without explicit `duration` and directly setting `Alpha`.

**Review by bdach — DISMISSED** [review]

File: osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs (hunk 3 above)
Comment (EVAST9919) [reply osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs]: Just easier to read imo, nothing special

File: osu.Game/Overlays/OverlayRulesetTabItem.cs (hunk 1 above)
Comment (EVAST9919) [reply osu.Game/Overlays/OverlayRulesetTabItem.cs]: Item deselecting happens after disabling, so we don't want to update the colour here. Tricky stuff, but my best thoughts so far, will revisit it

Comment (EVAST9919) [reply osu.Game/Overlays/OverlayRulesetTabItem.cs]: Okay, should be fixed

**Review by smoogipoo — APPROVED** [review]
lgtm 👍

**Pull request merged**